question-service/controller/question-controller.js

```diff
@@ -105,26 +119,35 @@ async function updateQuestionById(req, res) {
     try {
```

Comment:
Could we spend a little bit of time cleaning this function up? Perhaps by constructing an update object according to the input in the request, and update it accordingly. At the moment this function is slightly difficult to comprehend